cmd/s3-vpc-traffic-only/s3vpctrafficonly.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/configservice"
+	"strings"
+)
+
+type AWSConfigService interface {
+	PutEvaluations(*configservice.PutEvaluationsInput) (*configservice.PutEvaluationsOutput, error)
+}
+
+func main() {
+	lambda.Start(handleRequest)
+}
+
+func handleRequest(ctx context.Context, configEvent events.ConfigEvent) error {
+	cSession := session.Must(session.NewSession())
+	svc := configservice.New(cSession)
+	err := handleRequestWithConfigService(ctx, configEvent, svc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handleRequestWithConfigService(ctx context.Context, configEvent events.ConfigEvent, svc AWSConfigService) error {
+	var status string
+	var invokingEvent InvokingEvent
+	var configurationItem ConfigurationItem
+
+	invokingEvent, err := getInvokingEvent([]byte(configEvent.InvokingEvent))
+
+	if err != nil {
+		panic(err)
+	}
+
+	configurationItem = invokingEvent.ConfigurationItem
+
+	if params := getParams(configEvent, "excludeBuckets"); params != nil {
+		for _, v := range params {
+			if v == configurationItem.ResourceName {
+				fmt.Println("Skipping over Compliance check for resource", v, "Params: excludeBuckets")
+				status = "NOT_APPLICABLE"
+			}
+		}
+	}
+
+	if isApplicable(configurationItem, configEvent) && status == "" {
+		status = evaluateCompliance(configurationItem)
+	} else {
+		status = "NOT_APPLICABLE"
+	}
+
+	var evaluations []*configservice.Evaluation
+
+	evaluation := &configservice.Evaluation{
+		ComplianceResourceId:   aws.String(configurationItem.ResourceID),
+		ComplianceResourceType: aws.String(configurationItem.ResourceType),
+		ComplianceType:         aws.String(status),
+		OrderingTimestamp:      aws.Time(configurationItem.ConfigurationItemCaptureTime),
+	}
+
+	evaluations = append(evaluations, evaluation)
+	putEvaluations := &configservice.PutEvaluationsInput{
+		Evaluations: evaluations,
+		ResultToken: aws.String(configEvent.ResultToken),
+		TestMode:    aws.Bool(false),
+	}
+
+	_, err = svc.PutEvaluations(putEvaluations)
+
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err
+	}
+
+	return nil
+}
+
+func evaluateCompliance(c ConfigurationItem) string {
+	if c.ResourceType != "AWS::S3::Bucket" {
+		return "NOT_APPLICABLE"
+	}
+
+	pd := c.SupplementaryConfiguration.BucketPolicy.PolicyText
+
+	if pd != nil {
+		pds := pd.(string)
+		var pdf = new(PolicyDocument)
+		err := json.Unmarshal([]byte(pds), &pdf)
+		if err != nil {
+			fmt.Println("Error Unmarshal :", err)
+			return "NON_COMPLIANT"
+		}
+		if pdf.Statement[0].Condition.StringEquals.AwsSourceVpc != nil || pdf.Statement[0].Condition.ForAllValuesStringNotEquals.AwsSourceVpc != nil {
```

Comment:
check if pdf.Statement length is higher than 0 so that you don't have a runtime error if there is an empty statement

Reply:
Solverd here https://github.com/in4it/aws-config-compliance-checks/pull/7/commits/5029c9409b220fa10b85c83ec15ece42fe269498